qt/widgets/common/inc/MantidQtWidgets/Common/FitScriptGeneratorPresenter.h

```diff
@@ -0,0 +1,72 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "DllOption.h"
+#include "MantidAPI/MatrixWorkspace_fwd.h"
+#include "MantidQtWidgets/Common/IFitScriptGeneratorPresenter.h"
+#include "MantidQtWidgets/Common/IndexTypes.h"
+
+#include <string>
+#include <vector>
+
+#include <QStringList>
+
+namespace MantidQt {
+namespace MantidWidgets {
+
+class IFitScriptGeneratorModel;
+class IFitScriptGeneratorView;
+
+class EXPORT_OPT_MANTIDQT_COMMON FitScriptGeneratorPresenter
+    : public IFitScriptGeneratorPresenter {
+public:
+  FitScriptGeneratorPresenter(IFitScriptGeneratorView *view,
+                              IFitScriptGeneratorModel *model,
+                              QStringList const &workspaceNames = QStringList(),
+                              double startX = 0.0, double endX = 0.0);
+  ~FitScriptGeneratorPresenter() override;
+
+  void notifyPresenter(ViewEvent const &event) override;
+
+  void openFitScriptGenerator() override;
+
+private:
+  void handleRemoveClicked();
+  void handleAddWorkspaceClicked();
+  void handleStartXChanged();
+  void handleEndXChanged();
+
+  void setWorkspaces(QStringList const &workspaceNames, double startX,
+                     double endX);
+  void addWorkspaces(
+      std::vector<Mantid::API::MatrixWorkspace_const_sptr> const &workspaces,
+      std::vector<WorkspaceIndex> const &workspaceIndices);
+  void addWorkspace(std::string const &workspaceName, double startX,
+                    double endX);
+  void addWorkspace(Mantid::API::MatrixWorkspace_const_sptr const &workspace,
```

Comment:
dont need the const, it is in the type

Reply:
Using const for the `MatrixWorkspace_const_sptr` pointer seems to be the standard used elsewhere in mantid. Like in the `CompareWorkspace`, `Plus` and `Minus` algorithms https://github.com/mantidproject/mantid/blob/399333f590e97021afbd4311cf5a6632624e074b/Framework/Algorithms/inc/MantidAlgorithms/CompareWorkspaces.h#L112

Reply:
that seems daft